code/modules/antagonists/wizard/equipment/soulstone.dm

```diff
@@ -341,6 +360,58 @@
 		return FALSE
 	return TRUE
 
+/obj/item/soulstone/proc/init_shade(mob/living/carbon/human/victim, mob/user, message_user = FALSE, mob/shade_controller)
+	if(!shade_controller)
+		shade_controller = victim
+	victim.stop_sound_channel(CHANNEL_HEARTBEAT)
+	var/mob/living/simple_animal/shade/soulstone_spirit = new /mob/living/simple_animal/shade(src)
+	soulstone_spirit.AddComponent(/datum/component/soulstoned, src)
+	soulstone_spirit.name = "Shade of [victim.real_name]"
+	soulstone_spirit.real_name = "Shade of [victim.real_name]"
+	soulstone_spirit.key = shade_controller.key
+	soulstone_spirit.copy_languages(victim, LANGUAGE_MIND)//Copies the old mobs languages into the new mob holder.
+	if(user)
+		soulstone_spirit.copy_languages(user, LANGUAGE_MASTER)
+	soulstone_spirit.update_atom_languages()
+	soulstone_spirit.grant_all_languages(FALSE, FALSE, TRUE) //Grants omnitongue
+	if(user)
+		soulstone_spirit.faction |= "[REF(user)]" //Add the master as a faction, allowing inter-mob cooperation
+		if(IS_CULTIST(user))
+			soulstone_spirit.mind.add_antag_datum(/datum/antagonist/cult)
+
+	soulstone_spirit.cancel_camera()
+	update_appearance()
+	if(user)
+		if(IS_CULTIST(user))
+			to_chat(soulstone_spirit, span_bold("Your soul has been captured! \
+				You are now bound to the cult's will. Help them succeed in their goals at all costs."))
+		else if(role_check(user))
+			to_chat(soulstone_spirit, span_bold("Your soul has been captured! You are now bound to [user.real_name]'s will. \
+				Help [user.p_them()] succeed in [user.p_their()] goals at all costs."))
+		if(message_user)
+			to_chat(user, "[span_info("<b>Capture successful!</b>:")] [victim.real_name]'s soul has been ripped \
+				from [victim.p_their()] body and stored within [src].")
+
+	victim.dust(drop_items = TRUE)
+
+/obj/item/soulstone/proc/get_ghost_to_replace_shade(mob/living/carbon/victim, mob/user)
```

Comment:
i think it would be cool if you added dmdoc comments to the procs you updated (and renamed) :)